File: 1620-CheckIfArrayPairsAreDivisibleByK/1620-CheckIfArrayPairsAreDivisibleByK.cpp
// Last updated: 10/1/2025, 12:47:27 AM
class Solution {
public:
    bool canArrange(vector<int>& arr, int k) {
        unordered_map<int, int> remainderCount;

    

    for (int num : arr) {

        int remainder = ((num % k) + k) % k;

        remainderCount[remainder]++;

    }

    for (int i = 0; i <= k / 2; i++) {

        if (i == 0) {

            if (remainderCount[0] % 2 != 0) return false;

        } else {

            if (remainderCount[i] != remainderCount[k - i]) return false;

        }

    }

    return true;


}
};

File: 1620-CheckIfArrayPairsAreDivisibleByK/1620-CheckIfArrayPairsAreDivisibleByK.py
# Last updated: 10/1/2025, 12:47:23 AM
class Solution:
    def canArrange(self, arr: List[int], k: int) -> bool:
        remainderCount =  defaultdict(int)
        
        for num in arr:
            remainder = ((num%k)+k)%k
            remainderCount[remainder] +=1
            
        for i in range((k//2)+1):
            if i ==0:
                if remainderCount[0]%2!=0 :
                    return False
            else:
                if remainderCount[i]!=remainderCount[k-i]:
                    return False
                
        return True
        